hapistrano.cabal

```diff
@@ -35,22 +35,22 @@ library
   exposed-modules:     System.Hapistrano
   other-modules:       System.Hapistrano.Types
   build-depends:       base >= 4.5 && < 4.10
-                     , either
-                     , filepath
-                     , mtl
-                     , process
-                     , time-locale-compat
-                     , time
-                     , transformers
+                     , either >= 4.0 && < 4.6
+                     , filepath >= 1.3 && < 1.5
+                     , mtl >= 2.0 && < 3.0
+                     , process >= 1.2 && < 1.5
+                     , time-locale-compat >= 0.1 && < 1.0
+                     , time >= 1.4.1 && < 2.0
+                     , transformers > 0.3 && < 1.0
   ghc-options:         -Wall
   default-language:    Haskell2010
 
 executable hap
   hs-source-dirs:      app
   main-is:             Main.hs
-  build-depends:       base
+  build-depends:       base >= 4.5 && < 4.10
```

Comment:
This one for base can be removed as well. Since the library has specific version constraints, that should be enough.